### PR TITLE
2163 - Integrated API Catalog with Search

### PR DESCRIPTION
**Pull request by riginoommen**
# Resolves 2163

# Explain the feature
Integrated the API catalog with the search microservice

## Does this PR introduce a breaking change

No

## Screenshots

N/A

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?

### CODE REVIEW

File: packages/api-catalog-service/src/graph/namespace/updateSearch.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import { fetch } from 'undici';
```
Comment (deshmukhmayur) [inline packages/api-catalog-service/src/graph/namespace/updateSearch.ts:1]: Any specific reason why you're using `undici`?
Use `cross-fetch` instead. It has a significantly lesses package size.

File: packages/api-gateway-service/config.json [hunk 2]
```diff
@@ -28,5 +28,10 @@
     "name": "Lighthouse",
     "uri": "http://lighthouse-service:8080/graphql",
     "subscriptionsUri": "ws://lighthouse-service:8080/subscriptions"
+  },
```
Comment (deshmukhmayur) [inline packages/api-gateway-service/config.json:31]: Can you separate this into a new PR?

**Review by deshmukhmayur — CHANGES_REQUESTED** [review]

File: packages/api-catalog-service/src/graph/namespace/updateSearch.ts (hunk 1 above)
Comment (riginoommen) [reply packages/api-catalog-service/src/graph/namespace/updateSearch.ts:1]: undici was already used over there with many placed. so I reused it

File: packages/api-gateway-service/config.json (hunk 2 above)
Comment (riginoommen) [reply packages/api-gateway-service/config.json:31]: Yes

File: packages/api-catalog-service/src/graph/namespace/updateSearch.ts (hunk 1 above)
Comment (akhilmhdh) [reply packages/api-catalog-service/src/graph/namespace/updateSearch.ts:1]: Undici is used mainly for apollo-http data source. 

Also undici is more than just a fetch. It does lot of things caching and concurrent request etc.

Comment (deshmukhmayur) [reply packages/api-catalog-service/src/graph/namespace/updateSearch.ts:1]: Oh ohk. I thought it was added for just making fetch requests. If it's used as apollo data source, then it's fine.

File: packages/api-gateway-service/config.json (hunk 2 above)
Comment (riginoommen) [reply packages/api-gateway-service/config.json]: Done

**Review by deshmukhmayur — APPROVED** [review]

**Review by hybridx — APPROVED** [review]

**Pull request merged**